spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractMigrateUriRegistrySqlCommand.java

```diff
@@ -122,7 +123,12 @@ public boolean canHandleInJdbcTemplate() {
 					armd.setName(split[1]);
 					armd.setType(ApplicationType.valueOf(split[0]).ordinal());
 					armd.setUri(uri);
-					armd.setVersion(arc.getResourceVersion(arc.getResource(uri)));
+					try {
+						armd.setVersion(arc.getResourceVersion(arc.getResource(uri)));
+					}
+					catch (Exception e) {
+						logger.warn("Skipping URI_REGISTRY item {} migration with uri {} due to error", name, uri, e);
```

Comment:
Should this be ERROR level instead of warning? Also, can we say this requires the app to be re-registered with a version for the deployment in Skipper?

Reply:
we should explain the error better, but a resource that has no version would never be able to be deployed even in classic mode.  The error occurs at deployment time.

Reply:
`Skipping URI_REGISTRY item {} migration with uri {} due to lack of version number in uri.`

Reply:
I would leave it as a warning level.  That resource would never have been able to have been deployed in anycase.

Reply:
ok, will update the log message and merge.